File: ConcreteApp/app/src/main/java/com/example/brunovsiq/concreteapp/networking/RestApi.java
package com.example.brunovsiq.concreteapp.networking;

import com.example.brunovsiq.concreteapp.models.Repository;

import org.json.JSONObject;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface RestApi {


    //https://api.github.com/search/repositories?q=language:Java&sort=stars&page=1

    @GET("search/repositories?q=language:Java&sort=stars&page={pageNumber}")
    Call<JSONObject> listRepos(@Path("pageNumber") Integer page);
}

File: data/src/main/java/com/rafael/fernandes/data/network/RestApi.kt
package com.rafael.fernandes.data.network

import com.rafael.fernandes.domain.model.GitPullRequests
import com.rafael.fernandes.domain.model.GitRepositories
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface RestApi {
    @GET("search/repositories?q=language:Java")
    fun getOnlineRepositories(
        @Query("sort") sort: String,
        @Query("page") page: Int
    ): Single<GitRepositories>

    @GET("repos/{user}/{repository}/pulls?state=all")
    fun listPullRequests(
        @Path("user") user: String,
        @Path("repository") repository: String
    ): Single<List<GitPullRequests>>
}